source/Editor/Entities/Plugin_Hahaha.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using Celeste;
+using Microsoft.Xna.Framework;
+using Monocle;
+
+namespace Snowberry.Editor.Entities;
+
+[Plugin("hahaha")]
+public class Plugin_Hahaha : Entity {
+
+    [Option("ifset")] public string IfSet = "";
+    [Option("triggerLaughSfx")] public bool TriggerLaughSFX = false;
+
+    public override int MaxNodes => 1;
+
+    public override void Render() {
+        base.Render();
+
+        // This rendering is like, fucked somehow. No clue why. See discord for explanation
+        // https://discord.com/channels/920506235927793675/1144069604491149364/1145856399335772221
+        DrawHahahas(Position, 1);
+        if (Nodes.Count != 0) {
+            DrawHahahas(Nodes[0], 0.5f);
+        }
+    }
+
+    public override void HQRender() {
+        base.HQRender();
+
+        if (Nodes.Count != 0)
+            DrawUtil.DottedLine(Center, Nodes[0] + new Vector2(Width, Height) / 2f, Color.White, 4, 2);
+    }
+
+    protected override IEnumerable<Rectangle> Select() {
+        yield return RectOnRelative(new(30, 10), justify: new(0.5f, 0.5f));
+
+        if (Nodes.Count != 0)
```

Comment:
you don't need to check the number of nodes before iterating through them; the `foreach` loop will just do nothing if there are none